spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/SpanName.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2013-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ *
+ * Annotation to provide the name for the Span
+ *
+ * @author Marcin Grzejszczak
+ */
+@Target(ElementType.TYPE)
```

Comment:
constraints are fine with me.. what sort of classes would be annotated with
this? for example, if they are those that are passed to executor services
(runnables, etc). That's the thing we'd explain in javadoc and/or example,
test etc.